internal/commands/vaultsecrets/secrets/displayer.go

```diff
@@ -63,14 +69,30 @@ func (d *displayer) Payload() any {
 		return d.openAppSecretsPayload()
 	}
 
+	if d.previewStaticSecretVerions != nil {
+		return d.previewStaticSecretVerionsPayload()
+	}
+
 	if d.secrets == nil {
 		return nil
 	}
 	return d.secretsPayload()
 }
 
 func (d *displayer) FieldTemplates() []format.Field {
-	fields := []format.Field{
+	if d.openAppSecrets != nil {
+		return d.openAppSecretsFieldTemplate()
+	}
+
+	if d.previewStaticSecretVerions != nil {
```

Comment:
Is a single displayer valuable still easier? Or should there be a separate displayer for versions?

Also check out this way of doing displayers, maybe easier to just have a package of the different field templates and use this:

https://github.com/hashicorp/hcp/blob/main/internal/commands/projects/read.go#L70
https://github.com/hashicorp/hcp/blob/main/internal/commands/projects/projects.go#L15-L20

Reply:
Yeah, ideally we should abstract some of it. However, I am not planning to cross that bridge yet. I'll make a note on jira ticket with recommendation as we will need to revisit this when more things are added.